tests/test_sql_query.py

```diff
@@ -644,9 +632,28 @@ async def test_sql_schemas(self):
         with self.assertRaisesRegexTx(
             asyncpg.UndefinedTableError, "unknown table"
         ):
-            await self.squery(
-                '''
-                SET search_path TO my_module;
-                SELECT id FROM "Person";
-                '''
-            )
+            await self.scon.execute('SET search_path TO my_module;')
+            await self.scon.fetch('SELECT id FROM "Person";')\
+
+
+    async def test_sql_static_eval(self):
+        res = await self.squery_values('select current_schema;')
+        self.assertEqual(res, [['public']])
+
+        await self.scon.execute("set search_path to blah")
+        res = await self.squery_values("select current_schema")
+        self.assertEqual(res, [['blah']])
+
+        await self.squery_values('set search_path to blah;')
+        res = await self.squery_values('select current_schema;')
+        # self.assertEqual(res, [['blah']])
```

Comment:
Ah this is failing because of the compile cache ... the previous `select current_schema;` is cached as the statically compiled `select 'public'...` thing and will keep working that way. I'm thinking maybe we could inject the values in the I/O server instead of the compiler, or we'll have to disable the compile cache when it is statically evaluated (a `cachable` flag somewhere?).

Reply:
Update after the meeting:

1. It's not the compile cache (the cache key is a tuple of `hash(sql)` and the `fe_settings` which is just `search_path` for now), but rather a reused prepared statement from asyncpg.
2. I'll try to add a hack to re-compile on bind if `fe_settings` changed.

Reply:
3. If:
   - this becomes too hacky or
   - we more functions that depend on the connection state or
   - we want to support `current_schemas( expr_that_is_not_const )`
   ... this static eval approach is no good and we have to implement such functions in SQL and evaluate them in Postgres. The hard part is "how to pass connection state from our server to postgres". It could be done in two ways (but we don't have a consensus which is less painfull):
   - store the state in a temp table which has to be populated every time a backend connection is received from the pool,
   - figure out which connection state variables are used during resolving and pass them to Postgres as query parameters.